__lzcnt64 fix

`NativeJIT/inc/NativeJIT/BitOperations.h:137:35: error: use of undeclared identifier
      '__lzcnt64'
            *highestBitSetIndex = __lzcnt64(value);`

Fixed adding 
`#define __LZCNT__`
in BitOperations.h.

`[  PASSED  ] 108 tests`

 OSX Yosemite.

with `__LZCNT__` defined at the top of `BitOperations.h` I am getting something else:
```
In file included from /usr/lib/gcc/x86_64-linux-gnu/6/include/immintrin.h:73:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/6/include/x86intrin.h:48,
                 from /tmp/NativeJIT/inc/NativeJIT/BitOperations.h:37,
                 from /tmp/NativeJIT/inc/NativeJIT/CodeGen/X64CodeGenerator.h:32,
                 from /tmp/NativeJIT/inc/NativeJIT/CodeGen/FunctionBuffer.h:40,
                 from /tmp/NativeJIT/src/CodeGen/FunctionBuffer.cpp:26:
/usr/lib/gcc/x86_64-linux-gnu/6/include/lzcntintrin.h: In function ‘short unsigned int __lzcnt16(short unsigned int)’:
/usr/lib/gcc/x86_64-linux-gnu/6/include/lzcntintrin.h:41:29: error: ‘__builtin_clzs’ was not declared in this scope
   return __builtin_clzs (__X);
                             ^
src/CodeGen/CMakeFiles/CodeGen.dir/build.make:158: recipe for target 'src/CodeGen/CMakeFiles/CodeGen.dir/FunctionBuffer.cpp.o' failed

```
I am actually confused what this option is supposed to give.
I grepped the GCC headers for `__LZCNT__` and found only the following:
```
#ifndef __LZCNT__
#pragma GCC push_options
#pragma GCC target("lzcnt")
#define __DISABLE_LZCNT__
#endif /* __LZCNT__ */
```
And then it does not seem to be sensitive to `__DISABLE_LZCNT__` and it gets undefined at the end.

`clang-3.9` is also upset (even with `__LZCNT__` defined):
```
/tmp/NativeJIT/inc/NativeJIT/BitOperations.h:139:35: error: always_inline function '__lzcnt64'
      requires target feature 'lzcnt', but would be inlined into function 'GetHighestBitSet' that is compiled without support
      for 'lzcnt'
            *highestBitSetIndex = __lzcnt64(value);
```

Thus I replaced the `__lzcnt64(value)` call in `BitOperations.h` with the following binary search and then `make test` passes:
```
           *highestBitSetIndex = 64;
           if (value) {
               if (value & 0xFFFFFFFF00000000) { *highestBitSetIndex -=32; value >>= 32; }
               if (value & 0x00000000FFFF0000) { *highestBitSetIndex -=16; value >>= 16; }
               if (value & 0x000000000000FF00) { *highestBitSetIndex -= 8; value >>=  8; }
               if (value & 0x00000000000000F0) { *highestBitSetIndex -= 4; value >>=  4; }
               if (value & 0b0000000000001100) { *highestBitSetIndex -= 2; value >>=  2; }
               if (value & 0b0000000000000010) { *highestBitSetIndex -= 1; value >>=  1; }
               if (value) { *highestBitSetIndex -= 1; }
           }
```
Perhaps this could be surrounded with CPU checks to be more efficient on modern CPUs. Anyway, I don't know what I am doing :-)

--- a/inc/NativeJIT/BitOperations.h
+++ b/inc/NativeJIT/BitOperations.h
@@ -134,10 +134,15 @@ namespace NativeJIT
                    ? true
                    : false;
 #else
+    #ifdef __LZCNT__
             *highestBitSetIndex = __lzcnt64(value);
             bool retval = *highestBitSetIndex != 64;
             *highestBitSetIndex = 63 - *highestBitSetIndex;
             return retval;
+    #else
+            *highestBitSetIndex = 63 - __builtin_clzll(value);
+            return value != 0;
+    #endif
 #endif
         }
 

